### ISSUE TITLE
Build Error : FFmpeg 3.1v

### ISSUE THREAD

**Issue by dingpong** [Microsoft/FFmpegInterop#80]
Build Errors

1>src\ffmpeg\ffmpeginterop\source\ffmpeginteropmss.cpp(287): error C4996: 'AVStream::codec': was declared deprecated
1>src\ffmpeg\ffmpeg\build\windows10\x64\include\libavformat\avformat.h(880): note: see declaration of 'AVStream::codec'
1>src\ffmpeg\ffmpeginterop\source\ffmpeginteropmss.cpp(325): error C4996: 'AVStream::codec': was declared deprecated
1>src\ffmpeg\ffmpeg\build\windows10\x64\include\libavformat\avformat.h(880): note: see declaration of 'AVStream::codec'
1>src\ffmpeg\ffmpeginterop\source\uncompressedaudiosampleprovider.cpp(109): error C4996: 'avcodec_decode_audio4': was declared deprecated
1>src\ffmpeg\ffmpeg\build\windows10\x64\include\libavcodec\avcodec.h(4704): note: see declaration of 'avcodec_decode_audio4'
1>src\ffmpeg\ffmpeginterop\source\uncompressedvideosampleprovider.cpp(123): error C4996: 'avcodec_decode_video2': was declared deprecated
1>src\ffmpeg\ffmpeg\build\windows10\x64\include\libavcodec\avcodec.h(4753): note: see declaration of 'avcodec_decode_video2'

**Comment by dingpong** [comment]
I had to temporarily modify three functions.
[ffmpeg3_1v.txt](https://github.com/Microsoft/FFmpegInterop/files/336423/ffmpeg3_1v.txt)

**Comment by khouzam** [comment]
I've created a new branch, `helmut-steiner-ffmpeg3.1`, which has @helmut-steiner changes to incorporate FFmpeg 3.1 as well as the changes in logic for the new `avcodec_send_packet` and `avcodec_receive_frame` logic. I've tried it on a few files and things look good. According to the samples that we pull out of the file and the ones that we pull out of the decoder, everything matches up.

I need to do some more testing with a bigger variety of files, but I'd like to see if other can try it and see if they find any issues.

Thanks
